src/runtime/RegExpObject.cpp

```diff
@@ -497,6 +497,19 @@ ArrayObject* RegExpObject::createRegExpMatchedArray(ExecutionState& state, const
             }
         }
     }
+    if (m_yarrPattern->m_namedGroupToParenIndex.empty()) {
+        arr->defineOwnProperty(state, ObjectPropertyName(state.context()->staticStrings().groups), ObjectPropertyDescriptor(Value(), ObjectPropertyDescriptor::AllPresent));
+    } else {
+        Object* groups = new Object(state);
+        groups->setPrototype(state, Value(Value::Null));
+        std::map<String*, unsigned> orderedMap;
```

Comment:
Oops, I missed one thing.
`std::map<String*, unsigned> orderedMap` is never used in this function, so please remove it.

BTW would you please check again that if `m_namedGroupToParenIndexVector` can be removed?
It seems that `m_namedGroupToParenIndex` is exactly same as `m_namedGroupToParenIndexVector`.
If order is the matter, how about sorting the pair values by allocating a temporal vector at here?

Reply:
IMO you may exploit `m_captureGroupNames` for sorting because each element of this vector is inserted in the order.